src/components/Icon/Icon.tsx

```diff
@@ -46,31 +43,11 @@ export function Icon({source, color, backdrop, accessibilityLabel}: Props) {
     );
   }
 
-  if (color && !newDesignLanguage && isNewDesignLanguageColor(color)) {
-    // eslint-disable-next-line no-console
-    console.warn(
-      'You have selected a color meant to be used in the new design language but new design language is not enabled.',
-    );
-  }
-
-  if (
-    color &&
-    sourceType === 'external' &&
-    newDesignLanguage === true &&
-    isNewDesignLanguageColor(color)
-  ) {
-    // eslint-disable-next-line no-console
-    console.warn(
-      'Recoloring external SVGs is not supported with colors in the new design language. Set the intended color on your SVG instead.',
-    );
-  }
-
   const className = classNames(
     styles.Icon,
     color && styles[variationName('color', color)],
```

Comment:
Do we want people to still be able to specify color props like `white` and `orange` in the new DL or do we want to force people to use roles like `critical` and `warning`?

I'm thinking the latter.

If people agree we should remove  the strings in `Color` from IconProps over here https://github.com/Shopify/polaris-react/blob/master/src/types.ts#L57, and then you can remove all the css classes for `.colorWhite` `.colorOrange` etc. Also check if Color is used anywhere else and if not so remove that type.